07/AcyclicGraph.scala

```diff
@@ -0,0 +1,214 @@
+import scala.io.Source
+
+object AcyclicGraph {
+case class Worker(step: Step, startTick: Int)
+
+case class Dependency(name: String, satisfied: Boolean)
```

Comment:
`Boolean` flags on data elements can often better represented using external set of objects. Instead of having `satisfied == true` you can just check `setOrSatisfied.contains(name)`.